vela-templates/addons/fluxcd/definitions/helm-release.yaml

```
@@ -0,0 +1,77 @@
+apiVersion: core.oam.dev/v1beta1
+kind: ComponentDefinition
+metadata:
+  name: helm-release
+  namespace: vela-system
+  annotations:
+    definition.oam.dev/description: "helm release is a group of K8s resources from either git repository or helm repo"
+spec:
+  workload:
+    type: autodetects.core.oam.dev
+  schematic:
+    cue:
+      template: |
+        output: {
+        	apiVersion: "source.toolkit.fluxcd.io/v1beta1"
+        	metadata: {
+        		name: context.name
+        	}
+        	if parameter.source_type == "GitRepo" {
+        		kind: "GitRepository"
+        		spec: {
+        			url: parameter.repo_url
+        			ref:
+        				branch: parameter.branch
+        			interval: parameter.pull_interval
+        		}
+        	}
+        	if parameter.source_type == "HelmRepo" {
+        		kind: "HelmRepository"
+        		spec: {
+        			interval: parameter.pull_interval
+        			url:      parameter.repo_url
+        		}
+        	}
+        }
+
+        outputs: release: {
+        	apiVersion: "helm.toolkit.fluxcd.io/v2beta1"
+        	kind:       "HelmRelease"
+        	metadata: {
+        		name: context.name
+        	}
+        	spec: {
+        		interval: parameter.pull_interval
+        		chart: {
+        			spec: {
+        				chart: parameter.chart
+        				sourceRef: {
+        					if parameter.source_type == "GitRepo" {
+        						kind: "GitRepository"
+        					}
+        					if parameter.source_type == "HelmRepo" {
+        						kind: "HelmRepository"
+        					}
+        					name:      context.name
+        					namespace: context.namespace
+        				}
+        				interval: parameter.pull_interval
+        			}
+        		}
+        	}
+        }
+
+        parameter: {
+        	source_type: "GitRepo" | "HelmRepo"
+        	//+usage=The Git or Helm repository URL, accept HTTP/S or SSH address as git url.
+        	repo_url: string
+        	//+usage=The interval at which to check for repository and relese updates.
+        	pull_interval: *"5m" | string
+        	//+usage=1.The relative path to helm chart for git source. 2. chart name for helm resource
+        	chart: string
+        	if source_type == "GitRepo" {
+        		//+usage=The Git reference to checkout and monitor for changes, defaults to master branch.
+        		branch: *"master" | string
+        	}
```

```suggestion
        		//+usage=The Git reference to checkout and monitor for changes, defaults to master branch.
        		branch:? *"master" | string
```

作为缺省参数即可，不要在 parameter里面加`if`

---

done